### PR TITLE
Issue #394: Fix unit test - test_table_coverage

### LINKED ISSUE #394: Unit test failure
```
vendor/bin/phpunit --filter test_table_coverage
Moodle 3.11.8+ (Build: 20220722), 0fe025a14e5a458d15fa0e280cc068491a547167
Php: 7.4.30, mysqli: 5.7.31, OS: Linux 5.15.0-41-generic x86_64
PHPUnit 9.5.13 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)

Time: 00:01.075, Memory: 328.00 MB

There was 1 failure:

1) core_privacy\privacy\provider_test::test_table_coverage
The following tables with user fields must be covered with metadata providers: 
  - zoom_breakout_participants (userid)
```
And one error:

```

Moodle 3.11.8+ (Build: 20220722), 0fe025a14e5a458d15fa0e280cc068491a547167
Php: 7.4.30, mysqli: 5.7.31, OS: Linux 5.15.0-41-generic x86_64
PHPUnit 9.5.13 by Sebastian Bergmann and contributors.

There was 1 error:

1) mod_zoom\privacy\mod_zoom_provider_test::test_delete_data_for_users
dml_write_exception: Error writing to database (You can't specify target table 'phpu_zoom_meeting_participants' for update in FROM clause
DELETE FROM phpu_zoom_meeting_participants WHERE id IN (SELECT zmp.id
                  FROM phpu_zoom_meeting_participants zmp
                  JOIN phpu_zoom_meeting_details zmd ON zmd.id = zmp.detailsid
                  JOIN phpu_zoom z ON zmd.zoomid = z.id
                  JOIN phpu_modules m ON m.name = 'zoom'
                  JOIN phpu_course_modules cm ON z.id = cm.instance AND m.id = cm.module
                  JOIN phpu_context ctx
                    ON ctx.instanceid = cm.id
                   AND ctx.contextlevel = ?
                  WHERE ctx.id = ?
                    AND zmp.userid IN (?,?))
[array (
  0 => 70,
  1 => 121001,
  2 => '125000',
  3 => '125001',
)])

/var/www/site/lib/dml/moodle_database.php:489
/var/www/site/lib/dml/moodle_read_slave_trait.php:291
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1686
/var/www/site/mod/zoom/classes/privacy/provider.php:339
/var/www/site/mod/zoom/tests/privacy/mod_zoom_provider_test.php:303
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97
```

### PR DESCRIPTION
**Pull request by hdagheda**
Fixes #394

### CODE REVIEW

File: classes/privacy/provider.php [hunk 1]
```diff
@@ -335,8 +363,13 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
...
         $params = array_merge($inparams, ['contextid' => $context->id, 'modlevel' => CONTEXT_MODULE]);
-
-        $DB->delete_records_select('zoom_meeting_participants', "id IN ($sql)", $params);
+        if ($zooms = $DB->get_records_sql($sql, $params)) {
+            $zmpids = [];
+            foreach ($zooms as $zoom) {
+                $zmpids[] = $zoom->id;
+            }
+            $DB->delete_records_select('zoom_meeting_participants', "id IN (".implode(',', $zmpids).")");
+        }
```
Comment (jrchamp) [inline classes/privacy/provider.php:372]: They we can hopefully avoid pulling all of the data locally using this command:
```suggestion
        $DB->delete_records_select('zoom_meeting_participants', "userid $insql AND detailsid IN ($sql)", $params);
```

File: lang/en/zoom.php [hunk 2]
```diff
@@ -389,3 +389,5 @@
 $string['showmedia'] = 'Show Media section';
 $string['showmediaonview'] = 'Show Media section on meeting page';
 $string['showmedia_help'] = 'Enabling this option will show the Media section on the meeting activity page.';
+$string['privacy:metadata:zoom_breakout_participants'] = 'The database table to store a list of zoom meeting breakout rooms participants';
+$string['privacy:metadata:zoom_breakout_participants:userid'] = 'The id of the participant user';
```
Comment (jrchamp) [inline lang/en/zoom.php:393]: These should be moved up to the other privacy strings so they are sorted alphabetically.

File: tests/privacy/mod_zoom_provider_test.php [hunk 3]
```diff
@@ -151,13 +151,15 @@ public function test_get_metadata() {
...
-        $this->assertCount(3, $itemcollection);
+        $this->assertCount(4, $itemcollection);
         $table = reset($itemcollection);
         $table2 = $itemcollection[1];
         $table3 = $itemcollection[2];
+        $table4 = $itemcollection[3];
         $this->assertEquals('zoom_meeting_participants', $table->get_name());
         $this->assertEquals('zoom_meeting_details', $table2->get_name());
         $this->assertEquals('zoom_meeting_recordings_view', $table3->get_name());
+        $this->assertEquals('zoom_breakout_participants', $table4->get_name());
```
Comment (jrchamp) [inline tests/privacy/mod_zoom_provider_test.php:162]: There should probably be a couple more tests for $table4 similar to the other $table2 and $table3 tests below.

File: classes/privacy/provider.php [hunk 4]
```diff
@@ -350,6 +383,33 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
                   WHERE ctx.id = :contextid
                     AND zmrv.userid $insql";
```
Comment (jrchamp) [inline classes/privacy/provider.php:384]: Similar SQL change as before.

File: classes/privacy/provider.php [hunk 5]
```diff
@@ -350,6 +383,33 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
...
 
-        $DB->delete_records_select('zoom_meeting_recordings_view', "id IN ($sql)", $params);
+        if ($zooms = $DB->get_records_sql($sql, $params)) {
+            $zmrvids = [];
+            foreach ($zooms as $zoom) {
+                $zmrvids[] = $zoom->id;
+            }
+            $DB->delete_records_select('zoom_meeting_recordings_view', "id IN (".implode(',', $zmrvids).")");
+        }
```
Comment (jrchamp) [inline classes/privacy/provider.php:392]: Similar command change as before.

File: classes/privacy/provider.php [hunk 6]
```diff
@@ -350,6 +383,33 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
...
+                  FROM {zoom_breakout_participants} zbp
+                  JOIN {zoom_meeting_breakout_rooms} zmbr ON zmbr.id = zbp.breakoutroomid
+                  JOIN {zoom} z ON zmbr.zoomid = z.id
+                  JOIN {modules} m ON m.name = 'zoom'
+                  JOIN {course_modules} cm ON z.id = cm.instance AND m.id = cm.module
+                  JOIN {context} ctx
+                    ON ctx.instanceid = cm.id
+                   AND ctx.contextlevel = :modlevel
+                  WHERE ctx.id = :contextid
+                    AND zbp.userid $insql";
```
Comment (jrchamp) [inline classes/privacy/provider.php:404]: Similar SQL change as before.

File: classes/privacy/provider.php [hunk 7]
```diff
@@ -350,6 +383,33 @@ public static function delete_data_for_users(\core_privacy\local\request\approve
...
+
+        $params = array_merge($inparams, ['contextid' => $context->id, 'modlevel' => CONTEXT_MODULE]);
+        if ($zooms = $DB->get_records_sql($sql, $params)) {
+            $zbpids = [];
+            foreach ($zooms as $zoom) {
+                $zbpids[] = $zoom->id;
+            }
+            $DB->delete_records_select('zoom_breakout_participants', "id IN (".implode(',', $zbpids).")");
+        }
```
Comment (jrchamp) [inline classes/privacy/provider.php:413]: Similar command change as before.

**Review by jrchamp — COMMENTED** [review]
Sorry for the abbreviated review. I need to go, but I think I covered all of the items I wanted to mention. If you have any questions, please let me know. Thank you!

**Comment by jrchamp** [comment]
I'll add a commit with those suggestions and a fix for the CI tests.

**Comment by jrchamp** [comment]
@hdagheda Please verify that the Moodle tests pass on your system with the changes I've added.

**Comment by hdagheda** [comment]
Hi @jrchamp ,

I've tested and run tests locally, seems all good.
```
php vendor/bin/phpunit --filter mod_zoom_provider_test
Moodle 3.11.8+ (Build: 20220722), 417e0252f4e39b28c02d069f35ab5055c8ce39db
Php: 7.4.30, pgsql: 14.0 (Debian 14.0-1.pgdg110+1), OS: Linux 5.15.0-41-generic x86_64
PHPUnit 9.5.13 by Sebastian Bergmann and contributors.

.......                                                             7 / 7 (100%)

Time: 00:02.344, Memory: 328.00 MB

OK (7 tests, 39 assertions)
```

**Review by jrchamp — APPROVED** [review]
Thank you @hdagheda ! This is a great set of fixes.

**Pull request merged**